ps2.c

```diff
@@ -47,6 +47,18 @@ static int send_state = 0;
 int ps2_clk_out, ps2_data_out;
 int ps2_clk_in, ps2_data_in;
 
+#ifndef __GNUC__
```

Comment:
- __popcnt16, __popcnt or __popcnt64 for Visual Studio?
https://docs.microsoft.com/en-us/cpp/intrinsics/popcnt16-popcnt-popcnt64?view=vs-2019
- Keep it a macro and use uppercase?

Reply:
Not sure about this… docs say cpuid must be checked at runtime to ensure popcnt is supported.

Reply:
Please only keep the generic version. There is no point in optimizing this per compiler and making the code more complicated and more fragile. Besides, I bet most compilers will detect this pattern and generate the same code as `__builtin_parity(x)` anyway.

(Also: tabs, not spaces)

Reply:
Done!